In [24]:
# Dependencies
import pandas as pd
import plotly.express as px
import scipy.stats as sts
import os
from scipy.stats import linregress
from sklearn import datasets
import statsmodels.api as sm

Hypothesis:
If temperature affects the number of bikes crossing the Brooklyn Bridge, increasing the temperature to above 55 degrees  will lead to a higer number of bikes crossing the Brooklyn Bridge.
Null Hypothesis
If temperature does not affect the number of bikes crossing the Brooklyn Bridge, Higher High Temperature values will have no effect on the number of bikes crossing the Brooklyn Bridge.
Alternative Hypothesis
If temperature has an effect on the number of bikes crossing the Brooklyn Bridge, a temperature above 55 degrees will leead to increased number of bikes crossing the Brooklyn Bridge.

In [25]:
bikecounts_file = os.path.join('../','Resources', 'bikecounts.csv')
bikecounts_df = pd.read_csv(bikecounts_file)
bikecounts_df['Day'] = pd.to_datetime(bikecounts_df['Day'])
bikecounts_df.head(10)

Unnamed: 0        Date        Day  High Temp (°F)  Low Temp (°F)  \
0           0  2016-04-01 2016-04-01            78.1           66.0   
1           1  2016-04-02 2016-04-02            55.0           48.9   
2           2  2016-04-03 2016-04-03            39.9           34.0   
3           3  2016-04-04 2016-04-04            44.1           33.1   
4           4  2016-04-05 2016-04-05            42.1           26.1   
5           5  2016-04-06 2016-04-06            45.0           30.0   
6           6  2016-04-07 2016-04-07            57.0           53.1   
7           7  2016-04-08 2016-04-08            46.9           44.1   
8           8  2016-04-09 2016-04-09            43.0           37.9   
9           9  2016-04-10 2016-04-10            48.9           30.9   

  Precipitation  Brooklyn Bridge  Manhattan Bridge  Williamsburg Bridge  \
0          0.01           1704.0              3126               4115.0   
1          0.15            827.0              1646               2565.0   
2          0.09            526.0              1232               1695.0   
3      0.47 (S)            521.0              1067               1440.0   
4             0           1416.0              2617               3081.0   
5             0           1885.0              3329               3856.0   
6          0.09           1276.0              2581               3282.0   
7          0.01           1982.0              3455               4113.0   
8          0.09            504.0               997               1507.0   
9             0           1447.0              2387               3132.0   

   Queensboro Bridge  Total  
0             2552.0  11497  
1             1884.0   6922  
2             1306.0   4759  
3             1307.0   4335  
4             2357.0   9471  
5             2849.0  11919  
6             2457.0   9596  
7             3194.0  12744  
8             1502.0   4510  
9             2160.0   9126

In [26]:
print(bikecounts_df.describe())

       Unnamed: 0  High Temp (°F)  Low Temp (°F)  Brooklyn Bridge  \
count   30.000000       30.000000      30.000000        30.000000   
mean    14.500000       60.580000      46.413333      2269.633333   
std      8.803408       11.347289       9.662502       995.633241   
min      0.000000       39.900000      26.100000       504.000000   
25%      7.250000       55.500000      44.100000      1511.250000   
50%     14.500000       62.100000      46.900000      2379.500000   
75%     21.750000       68.000000      50.000000      3104.000000   
max     29.000000       81.000000      66.000000      3871.000000   

       Manhattan Bridge  Williamsburg Bridge  Queensboro Bridge         Total  
count         30.000000            30.000000          30.000000     30.000000  
mean        4049.533333          4862.466667        3352.866667  14534.500000  
std         1729.740976          1840.652747        1115.381263   5650.877227  
min          997.000000          1440.000000        1306.0

In [27]:
#High Temp higher than 55.0
#looking at the mean of bikes crossing the Brooklyn Bridge, when the High temperature is above 55.0, the average number of
#bikes is about 2680 compared to an average of 1183 for temperature below 55.0 as seen in the cell below.
Brooklyn=bikecounts_df.loc[bikecounts_df['High Temp (°F)']>55.0]
Brooklyn.describe()

Unnamed: 0  High Temp (°F)  Low Temp (°F)  Brooklyn Bridge  \
count   22.000000       22.000000      22.000000        22.000000   
mean    18.000000       66.022727      50.336364      2680.954545   
std      7.489675        7.352028       7.006565       758.836483   
min      0.000000       57.000000      39.900000      1045.000000   
25%     13.250000       60.600000      46.000000      2135.000000   
50%     18.500000       65.000000      48.000000      2819.000000   
75%     23.750000       70.825000      52.825000      3307.000000   
max     29.000000       81.000000      66.000000      3871.000000   

       Manhattan Bridge  Williamsburg Bridge  Queensboro Bridge         Total  
count         22.000000            22.000000          22.000000     22.000000  
mean        4761.636364          5658.409091        3819.409091  16920.409091  
std         1345.085890          1350.274951         834.874194   4241.714777  
min         2178.000000          2762.000000        2082.000000   8067.000000  
25%         3881.250000          4905.250000        3266.250000  13808.000000  
50%         4929.500000          5745.500000        3934.500000  17501.000000  
75%         5553.250000          6456.000000        4340.250000  19541.000000  
max         6951.000000          7834.000000        5032.000000  23318.000000

In [28]:
#High Temp lower than 55.0
Brooklyn_low=bikecounts_df.loc[bikecounts_df['High Temp (°F)']<55.0]
Brooklyn_low.describe()

Unnamed: 0  High Temp (°F)  Low Temp (°F)  Brooklyn Bridge  \
count    7.000000        7.000000       7.000000         7.000000   
mean     5.428571       44.271429      33.728571      1183.000000   
std      2.636737        3.009271       5.852838       656.521642   
min      2.000000       39.900000      26.100000       504.000000   
25%      3.500000       42.550000      30.450000       523.500000   
50%      5.000000       44.100000      33.100000      1416.000000   
75%      7.500000       45.950000      35.950000      1666.000000   
max      9.000000       48.900000      44.100000      1982.000000   

       Manhattan Bridge  Williamsburg Bridge  Queensboro Bridge         Total  
count          7.000000             7.000000           7.000000      7.000000  
mean        2154.857143          2689.142857        2096.428571   8123.428571  
std         1057.684646          1131.719846         757.659303   3589.465552  
min          997.000000          1440.000000        1306.000000   4335.000000  
25%         1149.500000          1601.000000        1404.500000   4634.500000  
50%         2387.000000          3081.000000        2160.000000   9126.000000  
75%         2973.000000          3494.000000        2603.000000  10695.000000  
max         3455.000000          4113.000000        3194.000000  12744.000000

In [29]:
#precipitation vs. number of bikes across Brooklyn Bridge
bikes=bikecounts_df['Brooklyn Bridge']
precipitation=bikecounts_df['Precipitation']
precipitation_v_bikes = px.scatter(x=precipitation, y=bikes)
precipitation_v_bikes.update_layout(
    xaxis_title="Precipitation",
    yaxis_title="Number of Bikes"
)

In [30]:
bikes=bikecounts_df['Brooklyn Bridge']
low=bikecounts_df['Low Temp (°F)']
low_v_bikes = px.scatter(x=low, y=bikes)
low_v_bikes.update_layout(
    xaxis_title="Low Temp",
    yaxis_title="Number of Bikes"
)

In [31]:
bikes=bikecounts_df['Brooklyn Bridge']
high=bikecounts_df['High Temp (°F)']
high_v_bikes = px.scatter(x=high, y=bikes)
high_v_bikes.update_layout(
    xaxis_title="High Temp",
    yaxis_title="Number of Bikes"
)

In [32]:
# Calculate the Pearson correlation coefficient for each pair of factors and show the scatter plot below it
print(f"The correlation between hight temp and the number of bikes crossing the Brooklyn Bridge is {sts.pearsonr(high, bikes)[0]}")
high_v_bikes.show()

The correlation between hight temp and the number of bikes crossing the Brooklyn Bridge is 0.7393765058774274


In [14]:
# Calculate the Pearson correlation coefficient for each pair of factors and show the scatter plot below it
print(f"The correlation between low temp and the number of bikes crossing the Brooklyn Bridge is {sts.pearsonr(low, bikes)[0]}")
low_v_bikes.show()

The correlation between low temp and the number of bikes crossing the Brooklyn Bridge is 0.464550105799598


In [15]:
x_values = bikecounts_df['High Temp (°F)']
y_values = bikecounts_df['Brooklyn Bridge']
(slope, intercept, rvalue, pvalue, stderr) = linregress(x_values, y_values)
regress_values = x_values * slope + intercept
line_eq = "y = " + str(round(slope, 2)) + "x + " + str(round(intercept, 2))
print(line_eq)

y = 64.87x + -1660.45


In [16]:
high_temp_v_bikecounts_plotly = px.scatter(
    bikecounts_df,
    x='High Temp (°F)',
    y='Brooklyn Bridge',
    trendline='ols',
    trendline_color_override='green'
)
high_temp_v_bikecounts_plotly.update_layout(
    xaxis_title='High Temp',
    yaxis_title='Bike Count'
)

In [17]:
results = px.get_trendline_results(high_temp_v_bikecounts_plotly)


In [18]:
Y = bikecounts_df['Brooklyn Bridge']
X = bikecounts_df['High Temp (°F)']
X = sm.add_constant(X)
model = sm.OLS(Y,X)
results = model.fit()
results.params

const            -1660.453908
High Temp (°F)      64.874335
dtype: float64

In [19]:
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:        Brooklyn Bridge   R-squared:                       0.547
Model:                            OLS   Adj. R-squared:                  0.530
Method:                 Least Squares   F-statistic:                     33.77
Date:                Sat, 27 Mar 2021   Prob (F-statistic):           3.05e-06
Time:                        01:33:59   Log-Likelihood:                -237.29
No. Observations:                  30   AIC:                             478.6
Df Residuals:                      28   BIC:                             481.4
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
const          -1660.4539    687.707     -2.

Results discussion
There is a strong positive correlation between High Temp (°F) and the bikes crossing Brooklyn Bridge. 
The correlation between hight temp and the number of bikes crossing the Brooklyn Bridge is 0.7393765058774274 which is high enough to show that High Temp affects the number of bikes. Increasing the High Temp level leads to an increase in the number of bikes crossing the bridge. 
Comparing a High Temp of above 55 °F to the temperature below 55°F, we can see that the mean of the number of bikes crossing the Brooklyn Bridge when the temperature is above 55°F is about 2680.954545, while when the temperature is below 55°F, the mean is 1183.000000. This means that the hypothesis that a temperature above 55°F would lead to an increase in the number of bikes crossing the Brooklyn Bridge is accepted, and the null hypothesis is rejected.